### PR TITLE
implement dpnp.mean

### PR DESCRIPTION
**Pull request by vtavana**
In this PR, `dpnp.mean` is implemented using its `dpctl` counterpart.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/dpnp_array.py [hunk 1]
```diff
@@ -915,10 +915,14 @@ def max(
 
         return dpnp.max(self, axis, out, keepdims, initial, where)
 
-    def mean(self, axis=None, **kwargs):
-        """Returns the average of the array elements."""
+    def mean(self, axis=None, dtype=None, keepdims=False, out=None, where=True):
```
Comment (antonwolfy) [inline dpnp/dpnp_array.py:918]: to align with numpy:
```suggestion
    def mean(self, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
```

File: tests/test_statistics.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 import dpnp
 
-from .helper import get_all_dtypes
+from .helper import assert_dtype_allclose, get_all_dtypes
```
Comment (antonwolfy) [inline tests/test_statistics.py:11]: Could you please add `test_statistics.py` and `test_meanvar.py` to `TEST_SCOPE` of `.github/workflows/conda-package.yml`?

**Review by antonwolfy — COMMENTED** [review]
We need also to secure performance against the previous implementation of `dpnp.mean` on PVC and Iris Xe. Could you please check that here is no performance degradation comes with the new implementation?

**Comment by vtavana** [comment]
Performance on PVC (Intel(R) Data Center GPU Max 1550):
Timing is obtained using `%timeit -r 7 -n 100` for three different cases:

1. `a = dpnp.arange(1024*32).reshape(1024,32)`

| Function | Old implementation | New Implementation |
| :---         |     :---:      |    :---:    |
| `mean(a)`   | 1.98 ms ± 128  µs    | 1.28 ms ± 44.4 µs   |
| `mean(a, axis=0)`   | 2.39 ms ± 26.6 µs       | 1.52 ms ± 907 µs      |
| `mean(a, axis=1)`   | 1.97 ms ± 269 µs      | 1.37 ms ± 494 µs      |

2.  `a = dpnp.arange(1024*32).reshape(32,1024)`

| Function | Old implementation | New Implementation |
| :---         |     :---:      |    :---:    |
| `mean(a)`   | 1.89 ms ± 150  µs    | 1.3 ms ± 33 µs   |
| `mean(a, axis=0)`   | 2.41 ms ± 48.3 µs       | 1.17 ms ± 34.1 µs      |
| `mean(a, axis=1)`   | 2.33 ms ± 135 µs      | 1.18 ms ± 26.9 µs      |

3.  `a = dpnp.arange(300*300).reshape(300,300)`

| Function | Old implementation | New Implementation |
| :---         |     :---:      |    :---:    |
| `mean(a)`   | 2.21 ms ± 78.3  µs    | 1.29 ms ± 58.5 µs   |
| `mean(a, axis=0)`   | 2.4 ms ± 28.9 µs       | 1.15 ms ± 29.8 µs      |
| `mean(a, axis=1)`   | 2.39 ms ± 30 µs      | 1.16 ms ± 34 µs      |

**Comment by vtavana** [comment]
Performance on Iris Xe (Intel(R) Graphics [0x46a8]):
Timing is obtained using `%timeit -r 7 -n 100` for three different cases:

1. `a = dpnp.arange(1024*32).reshape(1024,32)`

| Function | Old implementation | New Implementation |
| :---         |     :---:      |    :---:    |
| `mean(a)`   | 2.13 ms ± 265  µs    | 621 µs ± 93 µs   |
| `mean(a, axis=0)`   | 1.96 ms ± 127 µs       | 673 µs ± 102 µs      |
| `mean(a, axis=1)`   | 1.97 ms ± 122 µs      | 674 µs ± 71.4 µs      |

2.  `a = dpnp.arange(1024*32).reshape(32,1024)`

| Function | Old implementation | New Implementation |
| :---         |     :---:      |    :---:    |
| `mean(a)`   | 1.9 ms ± 101  µs    | 574 µs ± 87 µs   |
| `mean(a, axis=0)`   | 1.99 ms ± 55.3 µs       | 737 µs ± 98.2 µs      |
| `mean(a, axis=1)`   | 1.97 ms ± 105 µs      | 656 µs ± 85.1 µs      |

3.  `a = dpnp.arange(300*300).reshape(300,300)`

| Function | Old implementation | New Implementation |
| :---         |     :---:      |    :---:    |
| `mean(a)`   | 1.92 ms ± 147  µs    | 622 µs ± 74.5 µs   |
| `mean(a, axis=0)`   | 1.98 ms ± 113 µs       | 654 µs ± 103 µs      |
| `mean(a, axis=1)`   | 1.96 ms ± 108 µs      | 698 µs ± 122 µs      |

**Review by antonwolfy — APPROVED** [review]

**Pull request merged**